plugins/aws/resoto_plugin_aws/resource/athena.py

```diff
@@ -0,0 +1,227 @@
+from resotolib.types import Json
+from resoto_plugin_aws.aws_client import AwsClient
+from resoto_plugin_aws.utils import ToDict
+from typing import Type, cast
+import concurrent.futures
+
+
+@define(eq=False, slots=False)
+class AwsAthenaEncryptionConfiguration:
+    kind: ClassVar[str] = "aws_athena_encryption_configuration"
+    mapping: ClassVar[Dict[str, Bender]] = {"encryption_option": S("EncryptionOption"), "kms_key": S("KmsKey")}
+    encryption_option: Optional[str] = None
+    kms_key: Optional[str] = None
+
+
+@define(eq=False, slots=False)
+class AwsAthenaResultConfiguration:
+    kind: ClassVar[str] = "aws_athena_result_configuration"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "output_location": S("OutputLocation"),
+        "encryption_configuration": S("EncryptionConfiguration") >> Bend(AwsAthenaEncryptionConfiguration.mapping),
+        "expected_bucket_owner": S("ExpectedBucketOwner"),
+    }
+    output_location: Optional[str] = None
+    encryption_configuration: Optional[AwsAthenaEncryptionConfiguration] = None
+    expected_bucket_owner: Optional[str] = None
+
+
+@define(eq=False, slots=False)
+class AwsAthenaEngineVersion:
+    kind: ClassVar[str] = "aws_athena_engine_version"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "selected_engine_version": S("SelectedEngineVersion"),
+        "effective_engine_version": S("EffectiveEngineVersion"),
+    }
+    selected_engine_version: Optional[str] = None
+    effective_engine_version: Optional[str] = None
+
+
+@define(eq=False, slots=False)
+class AwsAthenaWorkGroupConfiguration:
+    kind: ClassVar[str] = "aws_athena_work_group_configuration"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "result_configuration": S("ResultConfiguration") >> Bend(AwsAthenaResultConfiguration.mapping),
+        "enforce_work_group_configuration": S("EnforceWorkGroupConfiguration"),
+        "publish_cloud_watch_metrics_enabled": S("PublishCloudWatchMetricsEnabled"),
+        "bytes_scanned_cutoff_per_query": S("BytesScannedCutoffPerQuery"),
+        "requester_pays_enabled": S("RequesterPaysEnabled"),
+        "engine_version": S("EngineVersion") >> Bend(AwsAthenaEngineVersion.mapping),
+    }
+    result_configuration: Optional[AwsAthenaResultConfiguration] = None
+    enforce_work_group_configuration: Optional[bool] = None
+    publish_cloud_watch_metrics_enabled: Optional[bool] = None
+    bytes_scanned_cutoff_per_query: Optional[int] = None
+    requester_pays_enabled: Optional[bool] = None
+    engine_version: Optional[AwsAthenaEngineVersion] = None
+
+
+@define(eq=False, slots=False)
+class AwsAthenaWorkGroup(AwsResource):
+    kind: ClassVar[str] = "aws_athena_work_group"
+    api_spec: ClassVar[AwsApiSpec] = AwsApiSpec("athena", "list-work-groups", "WorkGroups")
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("Name"),
+        "ctime": S("CreationTime"),
+        "name": S("Name"),
+        "state": S("State"),
```

Comment:
nitpick: can you add the correct `prop_prefix` to all properties that are not from the base resource?

Reply:
Yeah, good idea. Thanks!

Reply:
Yeah, good idea. Thanks!